trpcultivate_phenotypes/trpcultivate_phenotypes.services.yml

```
@@ -0,0 +1,14 @@
+# TripalCultivate Phenotypes Services.
+
+services:
+  trpcultivate_phenotypes.terms:
+    class: 'Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesTermsService'
+    arguments: ['@config.factory']
+  trpcultivate_phenotypes.ontology:
+    class: 'Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesOntologyService'
+    arguments: ['@config.factory']
+  trpcultivate_phenotypes.vocabulary:
+    class: 'Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesVocabularyService'
+  trpcultivate_phenotypes.database:
+    class: 'Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesDatabaseService'
```

To use dependency Injection in a service, we tell Drupal to grab the chado database connection from the container in the services.yml as follows. This allows the database connection to be a parameter of the __construct().

```suggestion
    class: 'Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesDatabaseService'
    arguments: ['@tripal_chado.database']
```